metadata-ingestion/src/datahub/emitter/mcp_builder.py

```diff
@@ -35,20 +37,38 @@ def guid(self) -> str:
         return str(md5_hash.hexdigest())
 
 
-@dataclasses.dataclass
 class PlatformKey(DatahubKey):
     platform: str
-    instance: Optional[str]
+    instance: Optional[str] = None
+    environment: Optional[str] = None
+
+    @root_validator(pre=True)
+    def check_instance_environment(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        # Instance and environment can't be provided at the same time
+        if values.get("instance") is not None and values.get("environment") is not None:
+            del values["environment"]
+        # if environment set but instance is not then use environment for instance to make backward compatible
+        elif values.get("environment") is not None and values.get("instance") is None:
+            values["instance"] = values["environment"]
+            del values["environment"]
+
+        return values
 
 
-@dataclasses.dataclass
 class DatabaseKey(PlatformKey):
     database: str
 
 
-@dataclasses.dataclass
 class SchemaKey(DatabaseKey):
-    schema: str
+    db_schema: str = Field(alias="schema")
```

Comment:
why change to db_schema?

Reply:
schema is an internal property of Pydantic and you can't use it. :( 
That's why I had to set an alias to workaround this.